keanu-project/src/main/java/io/improbable/keanu/tensor/intgr/Nd4jIntegerTensor.java

```diff
@@ -21,6 +21,10 @@
 
 public class Nd4jIntegerTensor implements IntegerTensor {
 
+    static {
+        System.setProperty("dtype", "double");
```

Comment:
it's probably worth adding a comment here about why we need this line

Reply:
The ND4J docs say that you can also set this with:

    Nd4j.setDataType(DataBuffer.Type.DOUBLE);

which is probably preferable

Reply:
Yeah that has to be added before every call to create a new INDArray due to ND4J defaulting to FLOAT. I've created a TypedINDArrayFactory that requires you specify the data buffer type before creation instead of letting ND4J default.